messaging/broker_test.go

```diff
@@ -216,6 +229,71 @@ func (b *Broker) Close() {
 	b.Broker.Close()
 }
 
+// ReadAll reads all available messages for a replica. Panic on error.
+func (b *Broker) MustReadAll(replicaID uint64) (a []*messaging.Message) {
+	// Read message from the replica.
+	var buf bytes.Buffer
+	go func() {
```

Comment:
We talked about this outside of the PR but I'll add it here for posterity. The `WriteTo()` function hangs indefinitely and streams out subscribed topic data so in order to get read it we have to kick it off in a separate goroutine and wait for the broker to write to the buffer before we can decode it.